src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -61,43 +67,86 @@ def _point_coordinate(value: Any) -> dict[str, Any] | object:
+    Given the registration options, resolve the documenttype URL to use.
+
+    :arg field: for which kind of upload the document type must be resolved.
+    :return: the resolved document type URL, if any. Empty string means that the upload
+      should be skipped.
     """
-    options = {
-        new_key: plugin_options[key_in_opts]
-        for new_key, key_in_opts in key_mapping.items()
-        if key_in_opts in plugin_options
-    }
-    return options
+    catalogue = options.get("catalogue")
+    match field:
+        case "submission_report":
+            description = options["iot_submission_report"]
+            url_ref = options.get("informatieobjecttype_submission_report", "")
+        case "submission_csv":
+            description = options["iot_submission_csv"]
+            url_ref = options.get("informatieobjecttype_submission_csv", "")
+        case "attachment":
+            description = options["iot_attachment"]
+            url_ref = options.get("informatieobjecttype_attachment", "")
+        case _:  # pragma: no cover
+            raise RuntimeError(f"Unhandled field '{field}'.")
+
+    # descriptions only work if a catalogue is provided to look up the document type
+    # inside it
+    if catalogue is None:
+        return url_ref
+
+    if not description:
+        return url_ref
+
+    # domain and rsin should not be empty, otherwise our validation is broken.
+    assert catalogue["domain"] and catalogue["rsin"]
+    assert submission.completed_on is not None
+
+    version_valid_on = datetime_in_amsterdam(submission.completed_on).date()
+    catalogus = catalogi_client.find_catalogus(**catalogue)
+    if catalogus is None:
+        raise RuntimeError(f"Could not resolve catalogue {catalogue}")
+
+    versions = catalogi_client.find_informatieobjecttypen(
+        catalogus=catalogus["url"],
+        description=description,
+        valid_on=version_valid_on,
+    )
+    if versions is None:
+        raise RuntimeError(
+            f"Could not find a document type with description '{description}' that is "
+            f"valid on {version_valid_on.isoformat()}."
+        )
+
+    version = versions[0]
+    return version["url"]
 
 
 def register_submission_pdf(
     submission: Submission,
     options: RegistrationOptions,
     documents_client: DocumentenClient,
+    catalogi_client: CatalogiClient,
 ) -> str:
-    if not options["informatieobjecttype_submission_report"]:
+    document_type = _resolve_documenttype(
+        "submission_report", options, submission, catalogi_client
+    )
+    if not document_type:
         return ""
 
     submission_report = SubmissionReport.objects.get(submission=submission)
-    submission_report_options = build_options(
-        options,
-        {
-            "informatieobjecttype": "informatieobjecttype_submission_report",
-            "organisatie_rsin": "organisatie_rsin",
-            "doc_vertrouwelijkheidaanduiding": "doc_vertrouwelijkheidaanduiding",
```

Comment:
Is `doc_vertrouwelijkheidaanduiding` no longer needed? I don't see it in the new `options`

Reply:
the "funny" thing with these options was that they were never specified anywhere, so they were also not being applied as an override. it was just copy-pasted from the ZGW API's.

AFAIK, only the file component allows you to override the confidentiality level, and the ZGW API's allow you to configure a default.